tools/mason/MasonArgParse.chpl

```diff
@@ -83,7 +81,25 @@ module MasonArgParse {
     var _opts:[0.._numOpts-1] string;
     // number of acceptable values to be present after argument is indicated
     var _numArgs:range;
+    // whether or not the user is required to choose a value for this option
+    var _required:bool;
+    // one or more default values to assign if opt is not entered by user
+    var _defaultValue:list(string);
+    proc init(name:string, numOpts:int, opts:[?argsD] string, numArgs:range, 
+              required=false, defaultValue=new list(string)){
+                _name=name;
+                _numOpts=numOpts;
+                _opts=opts;
+                _numArgs=numArgs;
+                _required=required;
+                _defaultValue=defaultValue;
 
+                // make sure that if we make an argument required no default set
+                assert(!(_required && _defaultValue.size > 0), 
+                        "Required options do not support default values");
+              }
```

Comment:
Let's indent the function body two spaces in from the `proc` and put the `}` at the same level as the `proc`. Also, let's include a space before the opening `{`.